src/components/ModalWithForm.js

```diff
@@ -5,6 +5,7 @@ export default class ModalWithForm extends Modal {
     super({ modalSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._form = this._modalElement.querySelector(".modal__form");
+    this._saveButton = this._modalElement.querySelector(".modal__button");
   }
 
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```